src/app/services/order.service.ts

```diff
@@ -1,49 +1,19 @@
 import { Injectable } from '@angular/core';
 import {Order} from "../model/order";
 import {HttpClient} from "@angular/common/http";
-import {Observable} from "rxjs";
-import {Product} from "../model/product";
-import {ProductIdQuantity} from "../model/ProductIdQuantity";
+import {ProductIdQuantity} from "../model/product-id-quantity";
+import {LoginService} from "./login.service";
+import {environment} from "../../environments/environment";
 
 @Injectable({
   providedIn: 'root'
 })
 export class OrderService {
-  private orderURL = "http://localhost:3000/orders";
 
-  private order : Order | undefined;
+  constructor(private http: HttpClient, private loginService: LoginService) { }
 
-  constructor(private http: HttpClient) { }
-
-  getOrder()
-  {
-    return this.order;
-  }
-
-  sendOrder(order: Order) : Observable<unknown> {
-    return this.http.post<Order>(this.orderURL, order);
-  }
-
-  updateOrder(product: Product) {
-    if (this.order) {
-      let found = false;
-      let prod: ProductIdQuantity;
-      for (let index in this.order.products) {
-        prod = this.order.products[index];
-        if (prod.productId == product.id) {
-          prod.quantity++;
-          found = true;
-        }
-      }
-      if(!found)
-      {
-        this.order.products.push({
-          productId: product.id,
-          quantity: 1
-        })
-      }
-    } else {
-      this.order = {customer: "doej", products: [{productId: product.id, quantity: 1}]}
-    }
+  sendOrder(products: ProductIdQuantity[]) {
+    const order : Order = { customer: this.loginService.getUsername(), products: products};
+    return this.http.post(environment.orderURL, order, {responseType: 'text'});
```

Comment:
Generally, I'd like you to keep in the environment only the endpoint for the backend. In this case, 'localhost:3000' and then in services append the endpoint itself. E.g., environment.backendUrl + '/orders' or \`${environment.backendUrl}/orders\`